common/data_refinery_common/models/attributes.py

```diff
@@ -0,0 +1,90 @@
+from django.db import models
+
+from data_refinery_common.models.ontology_term import OntologyTerm
+
+BOOL = "bool"
+INT = "int"
+FLOAT = "float"
+ONTOLOGY_TERM = "ont"
+
+
+class AbstractAttribute(models.Model):
+    """This is an abstract class that defines all of the properties of a single
+    attribute on either a sample or an experiment. We then subclass this to
+    associate attributes with either an experiment or a sample."""
+
+    name = models.ForeignKey("OntologyTerm", on_delete=models.CASCADE, related_name="+")
+    unit = models.ForeignKey("OntologyTerm", on_delete=models.CASCADE, related_name="+", null=True)
+    probability = models.FloatField(null=True)
+    source = models.TextField()
```

Comment:
Does this mean that every annotation will duplicate the text in source? If so, we should keep an eye on how many entries there are and how long the string is.

Reply:
About the "value of this metadata", an example of a piece of metadata would be `{'name': 'sex', 'value': 'male'}`. The question is if we always want them to report this value as an ontology term, or if there could be cases where they would report it as a string (maybe for different kinds of metadata).

In terms of the source, that is a good point. We can look into ways to intern the string so that the full text is only stored once.